ec/ecresource/extensionresource/request.go

```diff
@@ -0,0 +1,115 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package extensionresource
+
+import (
+	"os"
+
+	"github.com/elastic/cloud-sdk-go/pkg/api"
+	"github.com/elastic/cloud-sdk-go/pkg/api/apierror"
+	"github.com/elastic/cloud-sdk-go/pkg/client/extensions"
+	"github.com/elastic/cloud-sdk-go/pkg/models"
+	"github.com/elastic/cloud-sdk-go/pkg/multierror"
+	"github.com/go-openapi/runtime"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func readRequest(d *schema.ResourceData, client *api.API) (*extensions.GetExtensionOK, error) {
```

Comment:
These functions should be really moved to the [elastic cloud-sdk-go](https://github.com/elastic/cloud-sdk-go/tree/master/pkg/api), but we can do it in a follow-up PR. We generally tend to return the `.Payload` of these types since we don't use any of the wrapper types like this one,  the return should be: `models.Extension`.

Reply:
Thanks. fixed by 31a44fb